File: engine/src/main/java/org/jesse/game/content/tombsofamascut/AbstractTheatreNPC.java
package org.jesse.game.content.tombsofamascut;

import org.jesse.game.util.Direction;
import org.jesse.game.world.entity.Location;
import org.jesse.game.world.entity.npc.NPC;

public abstract class AbstractTheatreNPC extends NPC {
    public AbstractTheatreNPC(int id, Location tile, Direction facing, int radius) {
        super(id, tile, facing, radius);
    }
}

File: content/raids/tob/src/main/kotlin/com/zenyte/game/content/theatreofblood/AbstractTheatreNPC.kt
package com.zenyte.game.content.theatreofblood

import com.zenyte.game.util.Direction
import com.zenyte.game.world.entity.Location
import com.zenyte.game.world.entity.npc.NPC

/**
 * Base class for Theatre of Blood NPCs. Relocated from the ToA package
 * (com.zenyte.game.content.tombsofamascut), where it had squatted since the
 * NR codebase; its only subclass is ToB's TheatreNPC. Converted to Kotlin
 * per the touch-it-convert-it policy (tob's sourceset is Kotlin-only).
 * NOTE: tile is nullable — TheatreNPC passes Location? through.
 */
abstract class AbstractTheatreNPC(id: Int, tile: Location?, facing: Direction, radius: Int) :
    NPC(id, tile, facing, radius) {

    override fun isRaidNpcForTwistedBow(): Boolean {
        return true
    }

    override fun isVampyric(): Boolean {
        return true
    }
}
